feat(manager/helmfile): Support kustomize

nikolaik opened the pull request:
## Changes

This conditionally installs [kustomize](https://github.com/kubernetes-sigs/kustomize) when [kustomizations are used in a helmfile.yaml](https://helmfile.readthedocs.io/en/latest/advanced-features/#adhoc-kustomization-of-helm-charts). 

## Context

Helmfile calls out to a kustomize binary in those cases when running `helmfile deps` so this is needed to support that scenario.

This is a followup to #19865 

There is also a scenario where a helmfile can point to a local directory (e.g. `chart: ./local`) containing a file named `kustomization.yaml`. This case is not handled with this initial PR. Is it ok to go forward without handling that case?

This depends on https://github.com/containerbase/base/pull/748

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

viceice commented on lib/modules/manager/helmfile/artifacts.ts:
```diff
@@ -52,6 +56,14 @@ export async function updateArtifacts({
           toolName: 'helmfile',
           constraint: config.constraints?.helmfile,
         },
+        ...(needKustomize
+          ? [
+              {
+                toolName: 'kustomize',
+                constraint: config.constraints?.kustomize,
+              },
+            ]
+          : []),
```
please use a normal if condition after the default assignment.

viceice commented on lib/modules/manager/helmfile/utils.ts:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  for (const doc of docs) {
+    if (!(doc && is.array(doc.releases))) {
```
```suggestion
    if (!is.array(doc?.releases)) {
```
simplify

viceice commented on lib/modules/manager/helmfile/utils.ts:
```diff
@@ -0,0 +1,50 @@
+}
+
+/** Looks for kustomize specific keys in a helmfile and returns true if found */
+export function areKustomizationsUsed(
```
i think we should do this at extraction time and store inside `managerData` field, so we don't need to parse files twice.

nikolaik replied:
Of course, that is much cleaner, updated the PR now! PS: @viceice Thanks again for the timely and helpful reviews <3

viceice commented on lib/util/exec/containerbase.ts:
```diff
@@ -80,6 +80,11 @@ const allToolConfig: Record<string, ToolConfig> = {
+  kustomize: {
+    datasource: 'github-releases',
+    packageName: 'kubernetes-sigs/kustomize',
+    versioning: semverVersioningId,
```
we need `extractVersion` here too, otherwise we get wrong releases

viceice commented on lib/modules/manager/helmfile/utils.ts:
```diff
@@ -0,0 +1,8 @@
+
+/** Returns true if kustomize specific keys exist in a helmfile release */
+export function areKustomizationsUsed(release: Release): boolean {
+  return Boolean(
```
```suggestion
  return (
```
no

viceice reviewed (commented):
please do a real repo test

nikolaik commented:
> please do a real repo test

Set up a test repo at https://github.com/nikolaik/test-renovate-helmfile-kustomize and it looks promising:  
```console
$ yarn start --token ... nikolaik/test-renovate-helmfile-kustomize --binary-source docker
(...)
$ node -r ts-node/register/transpile-only -- lib/renovate.ts --token ... nikolaik/test-renovate-helmfile-kustomize --binary-source docker
 INFO: Repository started (repository=nikolaik/test-renovate-helmfile-kustomize)
       "renovateVersion": "0.0.0-semantic-release"
 INFO: Dependency extraction complete (repository=nikolaik/test-renovate-helmfile-kustomize, baseBranch=main)
       "stats": {
         "managers": {"helmfile": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
 INFO: Branch updated (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "commitSha": "ac67b9d48519db81cb98de8abb1b39124fdccac9"
 INFO: PR updated (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "pr": 1,
       "prTitle": "Update Helm release raw to v0.2.5"
 INFO: Repository finished (repository=nikolaik/test-renovate-helmfile-kustomize)
       "cloned": true,
       "durationMs": 33217
✨  Done in 40.73s.
```

Creating this PR https://github.com/nikolaik/test-renovate-helmfile-kustomize/pull/1

viceice commented:
> > please do a real repo test
> 
> Set up a test repo at https://github.com/nikolaik/test-renovate-helmfile-kustomize and it looks promising:  
> ```console
> $ yarn start --token ... nikolaik/test-renovate-helmfile-kustomize --binary-source docker
> (...)
> $ node -r ts-node/register/transpile-only -- lib/renovate.ts --token ... nikolaik/test-renovate-helmfile-kustomize --binary-source docker
>  INFO: Repository started (repository=nikolaik/test-renovate-helmfile-kustomize)
>        "renovateVersion": "0.0.0-semantic-release"
>  INFO: Dependency extraction complete (repository=nikolaik/test-renovate-helmfile-kustomize, baseBranch=main)
>        "stats": {
>          "managers": {"helmfile": {"fileCount": 1, "depCount": 1}},
>          "total": {"fileCount": 1, "depCount": 1}
>        }
>  INFO: Branch updated (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
>        "commitSha": "ac67b9d48519db81cb98de8abb1b39124fdccac9"
>  INFO: PR updated (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
>        "pr": 1,
>        "prTitle": "Update Helm release raw to v0.2.5"
>  INFO: Repository finished (repository=nikolaik/test-renovate-helmfile-kustomize)
>        "cloned": true,
>        "durationMs": 33217
> ✨  Done in 40.73s.
> ```
> 
> Creating this PR https://github.com/nikolaik/test-renovate-helmfile-kustomize/pull/1

you should run with debug log level, so we see new commands. so tick rebase checkbox and run again

nikolaik commented:
Here is the debug run. By the looks of it, it seems to be installing the kustomize tool like it is supposed to 😊

```console
$ LOG_LEVEL=debug yarn start --token ... nikolaik/test-renovate-helmfile-kustomize --binary-source docker
(...)
$ node -r ts-node/register/transpile-only -- lib/renovate.ts --token ... nikolaik/test-renovate-helmfile-kustomize --binary-source docker
DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /home/nikolark/dev/renovate/config.js
DEBUG: No config file found on disk - skipping
DEBUG: File config
       "config": {}
DEBUG: CLI config
       "config": {
         "repositories": ["nikolaik/test-renovate-helmfile-kustomize"],
         "binarySource": "docker",
         "token": "***********"
       }
DEBUG: Env config
       "config": {"hostRules": []}
DEBUG: Combined config
       "config": {
         "hostRules": [],
         "repositories": ["nikolaik/test-renovate-helmfile-kustomize"],
         "binarySource": "docker",
         "token": "***********"
       }
DEBUG: Enabling forkProcessing while in non-autodiscover mode
DEBUG: Found valid git version: 2.39.2
DEBUG: Using default github endpoint: https://api.github.com/
DEBUG: GET https://api.github.com/user/emails = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=286)
DEBUG: Cannot read user/emails endpoint on GitHub to retrieve gitAuthor
DEBUG: Platform config
       "platformConfig": {
         "hostType": "github",
         "endpoint": "https://api.github.com/",
         "isGHApp": false,
         "isGhe": false,
         "userDetails": {"username": "nikolaik", "name": "..."},
         "userEmail": null
       },
       "renovateUsername": "nikolaik"
DEBUG: Adding token authentication for api.github.com to hostRules
DEBUG: Using baseDir: /tmp/renovate
DEBUG: Using cacheDir: /tmp/renovate/cache
DEBUG: Using containerbaseDir: /tmp/renovate/cache/containerbase
DEBUG: Initializing Renovate internal cache into /tmp/renovate/cache/renovate/renovate-cache-v1
DEBUG: Commits limit = null
DEBUG: Setting global hostRules
DEBUG: Adding token authentication for api.github.com to hostRules
DEBUG: validatePresets()
DEBUG: Reinitializing hostRules for repo
DEBUG: Clearing hostRules
DEBUG: Adding token authentication for api.github.com to hostRules
DEBUG: No dangling containers to remove
 INFO: Repository started (repository=nikolaik/test-renovate-helmfile-kustomize)
       "renovateVersion": "0.0.0-semantic-release"
DEBUG: Using localDir: /tmp/renovate/repos/github/nikolaik/test-renovate-helmfile-kustomize (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: PackageFiles.clear() - Package files deleted (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: initRepo("nikolaik/test-renovate-helmfile-kustomize") (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: nikolaik/test-renovate-helmfile-kustomize default branch = main (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using personal access token for git init (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Resetting npmrc (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: checkOnboarding() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: isOnboarded() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: findFile(renovate.json) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Initializing git repository into /tmp/renovate/repos/github/nikolaik/test-renovate-helmfile-kustomize (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Performing blobless clone (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: git clone completed (repository=nikolaik/test-renovate-helmfile-kustomize)
       "durationMs": 1152
DEBUG: latest repository commit (repository=nikolaik/test-renovate-helmfile-kustomize)
       "latestCommit": {
         "hash": "acfea149bebe380e16c936cc29a8ede9d398ba23",
         "date": "2023-03-13T12:24:53+01:00",
         "message": "initial commit",
         "refs": "HEAD -> main, origin/main, origin/HEAD",
         "body": "",
         "author_name": "...",
         "author_email": "..."
       }
DEBUG: Config file exists, fileName: renovate.json (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Retrieving issueList (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Retrieved 1 issues (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Repo is onboarded (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Found renovate.json config file (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Repository config (repository=nikolaik/test-renovate-helmfile-kustomize)
       "fileName": "renovate.json",
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["config:base"]
       }
DEBUG: migrateAndValidate() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: No config migration necessary (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: massaged config (repository=nikolaik/test-renovate-helmfile-kustomize)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["config:base"]
       }
DEBUG: migrated config (repository=nikolaik/test-renovate-helmfile-kustomize)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["config:base"]
       }
DEBUG: Setting hostRules from config (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Found repo ignorePaths (repository=nikolaik/test-renovate-helmfile-kustomize)
       "ignorePaths": [
         "**/node_modules/**",
         "**/bower_components/**",
         "**/vendor/**",
         "**/examples/**",
         "**/__tests__/**",
         "**/test/**",
         "**/tests/**",
         "**/__fixtures__/**"
       ]
DEBUG: No vulnerability alerts found (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: No vulnerability alerts found (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: findIssue(Dependency Dashboard) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Found issue 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: No baseBranches (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: extract() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Setting current branch to main (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: latest commit (repository=nikolaik/test-renovate-helmfile-kustomize)
       "branchName": "main",
       "latestCommitDate": "2023-03-13T12:24:53+01:00"
DEBUG: Using file match: (^|/)tasks/[^/]+\.ya?ml$ for manager ansible (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)requirements\.ya?ml$ for manager ansible-galaxy (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)galaxy\.ya?ml$ for manager ansible-galaxy (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.tool-versions$ for manager asdf (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: azure.*pipelines?.*\.ya?ml$ for manager azure-pipelines (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)batect(-bundle)?\.yml$ for manager batect (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)batect$ for manager batect-wrapper (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)WORKSPACE(|\.bazel)$ for manager bazel (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.bzl$ for manager bazel (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.bazelversion$ for manager bazelisk (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.?bitbucket-pipelines\.ya?ml$ for manager bitbucket-pipelines (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: buildkite\.ya?ml for manager buildkite (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.buildkite/.+\.ya?ml$ for manager buildkite (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Gemfile$ for manager bundler (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.cake$ for manager cake (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Cargo\.toml$ for manager cargo (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.circleci/config\.yml$ for manager circleci (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)cloudbuild\.ya?ml for manager cloudbuild (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Podfile$ for manager cocoapods (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)([\w-]*)composer\.json$ for manager composer (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)conanfile\.(txt|py)$ for manager conan (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)(?:deps|bb)\.edn$ for manager deps-edn (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)(?:docker-)?compose[^/]*\.ya?ml$ for manager docker-compose (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/|\.)(Docker|Container)file$ for manager dockerfile (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)(Docker|Container)file[^/]*$ for manager dockerfile (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.drone\.yml$ for manager droneci (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)fleet\.ya?ml for manager fleet (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)flux-system/(?:.+/)?gotk-components\.yaml$ for manager flux (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.fvm/fvm_config\.json$ for manager fvm (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.gitmodules$ for manager git-submodules (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: ^(workflow-templates|\.github/workflows)/[^/]+\.ya?ml$ for manager github-actions (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)action\.ya?ml$ for manager github-actions (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci-include (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)go\.mod$ for manager gomod (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)gradle\.properties$ for manager gradle (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)gradle/.+\.toml$ for manager gradle (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.versions\.toml$ for manager gradle (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)versions.props$ for manager gradle (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)versions.lock$ for manager gradle (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)gradle/wrapper/gradle-wrapper\.properties$ for manager gradle-wrapper (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)requirements\.yaml$ for manager helm-requirements (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)values\.yaml$ for manager helm-values (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)helmfile\.yaml$ for manager helmfile (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Chart\.yaml$ for manager helmv3 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)bin/hermit$ for manager hermit (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: ^Formula/[^/]+[.]rb$ for manager homebrew (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.html?$ for manager html (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)plugins\.(txt|ya?ml)$ for manager jenkins (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)jsonnetfile\.json$ for manager jsonnet-bundler (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: ^.+\.main\.kts$ for manager kotlin-script (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)kustomization\.ya?ml$ for manager kustomize (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)project\.clj$ for manager leiningen (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/|\.)pom\.xml$ for manager maven (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: ^(((\.mvn)|(\.m2))/)?settings\.xml$ for manager maven (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|\/).mvn/wrapper/maven-wrapper.properties$ for manager maven-wrapper (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)package\.js$ for manager meteor (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Mintfile$ for manager mint (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)mix\.exs$ for manager mix (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)flake\.nix$ for manager nix (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.node-version$ for manager nodenv (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)package\.json$ for manager npm (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.(?:cs|fs|vb)proj$ for manager nuget (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.(?:props|targets)$ for manager nuget (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)dotnet-tools\.json$ for manager nuget (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)global\.json$ for manager nuget (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.nvmrc$ for manager nvm (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)src/main/features/.+\.json$ for manager osgi (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)([\w-]*)requirements\.(txt|pip)$ for manager pip_requirements (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)setup\.py$ for manager pip_setup (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Pipfile$ for manager pipenv (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager poetry (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.pre-commit-config\.yaml$ for manager pre-commit (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)pubspec\.ya?ml$ for manager pub (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Puppetfile$ for manager puppet (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.python-version$ for manager pyenv (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.ruby-version$ for manager ruby-version (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.sbt$ for manager sbt (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: project/[^/]*\.scala$ for manager sbt (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: project/build\.properties$ for manager sbt (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)setup\.cfg$ for manager setup-cfg (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)Package\.swift for manager swift (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.tf$ for manager terraform (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.terraform-version$ for manager terraform-version (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)terragrunt\.hcl$ for manager terragrunt (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.terragrunt-version$ for manager terragrunt-version (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: \.tflint\.hcl$ for manager tflint-plugin (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: ^\.travis\.yml$ for manager travis (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: (^|/)\.vela\.ya?ml$ for manager velaci (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Using file match: ^\.woodpecker(?:/[^/]+)?\.ya?ml$ for manager woodpecker (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Matched 1 file(s) for manager helmfile: helmfile.yaml (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Matched 1 file(s) for manager kustomize: chart/kustomization.yaml (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: repositories discovered. (repository=nikolaik/test-renovate-helmfile-kustomize)
       "registryAliases": {"incubator": "https://charts.helm.sh/incubator/"}
DEBUG: manager extract durations (ms) (repository=nikolaik/test-renovate-helmfile-kustomize)
       "managers": {"helmfile": 3, "kustomize": 4}
DEBUG: Found helmfile package files (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Found 1 package file(s) (repository=nikolaik/test-renovate-helmfile-kustomize)
 INFO: Dependency extraction complete (repository=nikolaik/test-renovate-helmfile-kustomize, baseBranch=main)
       "stats": {
         "managers": {"helmfile": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: PackageFiles.add() - Package file saved for base branch (repository=nikolaik/test-renovate-helmfile-kustomize, baseBranch=main)
DEBUG: Package releases lookups complete (repository=nikolaik/test-renovate-helmfile-kustomize, baseBranch=main)
DEBUG: branchifyUpgrades (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: detectSemanticCommits() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: getCommitMessages (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: semanticCommits: detected "unknown" (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: semanticCommits: disabled (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: 1 flattened updates found: raw (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Returning 1 branch(es) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: config.repoIsOnboarded=true (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: packageFiles with updates (repository=nikolaik/test-renovate-helmfile-kustomize, baseBranch=main)
       "config": {
         "helmfile": [
           {
             "deps": [
               {
                 "depName": "raw",
                 "currentValue": "0.1.0",
                 "registryUrls": ["https://charts.helm.sh/incubator/"],
                 "managerData": {"needKustomize": true},
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "0.2.5",
                     "newValue": "0.2.5",
                     "releaseTimestamp": "2020-11-14T12:58:41.036Z",
                     "newMajor": 0,
                     "newMinor": 2,
                     "updateType": "minor",
                     "branchName": "renovate/raw-0.x"
                   }
                 ],
                 "packageName": "raw",
                 "warnings": [],
                 "versioning": "helm",
                 "sourceUrl": "https://github.com/helm/charts",
                 "registryUrl": "https://charts.helm.sh/incubator",
                 "homepage": "https://github.com/helm/charts/blob/master/incubator/raw",
                 "currentVersion": "0.1.0",
                 "isSingleVersion": true,
                 "fixedVersion": "0.1.0"
               }
             ],
             "datasource": "helm",
             "packageFile": "helmfile.yaml"
           }
         ]
       }
DEBUG: detectSemanticCommits() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: semanticCommits: returning "disabled" from cache (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: processRepo() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Processing 1 branch: renovate/raw-0.x (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Calculating hourly PRs remaining (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: getPrList success (repository=nikolaik/test-renovate-helmfile-kustomize)
       "pullsTotal": 1,
       "requestsTotal": 1,
       "apiQuotaAffected": true
DEBUG: currentHourStart=2023-03-13T20:00:00.000+01:00 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: PR hourly limit remaining: 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Calculating prConcurrentLimit (10) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: getBranchPr(renovate/raw-0.x) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: findPr(renovate/raw-0.x, undefined, open) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Found PR #1 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: 1 PRs are currently open (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: PR concurrent limit remaining: 9 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Calculated maximum PRs remaining this run: 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: PullRequests limit = 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Calculating hourly PRs remaining (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: currentHourStart=2023-03-13T20:00:00.000+01:00 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: PR hourly limit remaining: 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Calculating branchConcurrentLimit (10) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: 1 already existing branches found: renovate/raw-0.x (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Branch concurrent limit remaining: 9 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Calculated maximum branches remaining this run: 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Branches limit = 2 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: syncBranchState() (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: syncBranchState(): Branch cache not found, creating minimal branchState (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: getBranchPr(renovate/raw-0.x) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: findPr(renovate/raw-0.x, undefined, open) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Found PR #1 (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: branchExists=true (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: dependencyDashboardCheck=undefined (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: PR rebase requested=true (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Checking if PR has been edited (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: branch.isModified(): using git to calculate (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: branch.isModified() = false (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Found existing branch PR (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Checking schedule(at any time, null) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: No schedule defined (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Manual rebase requested via Dependency Dashboard (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Using reuseExistingBranch: false (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Setting current branch to main (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: latest commit (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "branchName": "main",
       "latestCommitDate": "2023-03-13T12:24:53+01:00"
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=false (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Starting search at index 130 (repository=nikolaik/test-renovate-helmfile-kustomize, packageFile=helmfile.yaml, branch=renovate/raw-0.x)
       "depName": "raw"
DEBUG: Found match at index 130 (repository=nikolaik/test-renovate-helmfile-kustomize, packageFile=helmfile.yaml, branch=renovate/raw-0.x)
       "depName": "raw"
DEBUG: repositories discovered. (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "registryAliases": {"incubator": "https://charts.helm.sh/incubator/"}
DEBUG: Contents updated (repository=nikolaik/test-renovate-helmfile-kustomize, packageFile=helmfile.yaml, branch=renovate/raw-0.x)
       "depName": "raw"
DEBUG: Setting CONTAINERBASE_CACHE_DIR to /tmp/renovate/cache/containerbase (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Using docker to execute (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "image": "sidecar"
DEBUG: Resolved stable matching version (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "toolName": "helm",
       "constraint": undefined,
       "resolvedVersion": "v3.11.2"
DEBUG: Resolved stable matching version (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "toolName": "helmfile",
       "constraint": undefined,
       "resolvedVersion": "v0.151.0"
DEBUG: Resolved stable matching version (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "toolName": "kustomize",
       "constraint": undefined,
       "resolvedVersion": "5.0.0"
DEBUG: containerbaseDir is inside cacheDir (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Resolved tag constraint (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "image": "docker.io/containerbase/sidecar"
DEBUG: Fetching Docker image: docker.io/containerbase/sidecar (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Finished fetching Docker image docker.io/containerbase/sidecar@sha256:0cd57e0174bb71ed07b039d794558841a9d126948cc54bff139e97a0cc51416d (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Executing command (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "command": "docker run --rm --name=renovate_sidecar --label=renovate_child -v \"/tmp/renovate/repos/github/nikolaik/test-renovate-helmfile-kustomize\":\"/tmp/renovate/repos/github/nikolaik/test-renovate-helmfile-kustomize\" -v \"/tmp/renovate/cache\":\"/tmp/renovate/cache\" -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w \"/tmp/renovate/repos/github/nikolaik/test-renovate-helmfile-kustomize\" docker.io/containerbase/sidecar bash -l -c \"install-tool helm v3.11.2 && install-tool helmfile v0.151.0 && install-tool kustomize 5.0.0 && helmfile deps -f helmfile.yaml\""
DEBUG: exec completed (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "durationMs": 7666,
       "stdout": "installing v2 tool helm v3.11.2\nlinking tool helm v3.11.2\nversion.BuildInfo{Version:\"v3.11.2\", GitCommit:\"912ebc1cd10d38d340f048efaf0abda047c3468e\", GitTreeState:\"clean\", GoVersion:\"go1.18.10\"}\nInstalled v2 /usr/local/buildpack/tools/v2/helm.sh in 1 seconds\nskip cleanup, not a docker build: 0b59a18ab905\ninstalling v2 tool helmfile v0.151.0\nlinking tool helmfile v0.151.0\n\n▓▓▓ /opt/buildpack/tools/helmfile/0.151.0/bin/helmfile\n\n  Version              0.151.0\n  Git Commit           f629ee1\n  Build Date           17 Feb 23 15:02 UTC (3 weeks ago)\n  Commit Date          17 Feb 23 06:09 UTC (3 weeks ago)\n  Dirty Build          no\n  Go version           1.20.1\n  Compiler             gc\n  Platform             linux/amd64\nInstalled v2 /usr/local/buildpack/tools/v2/helmfile.sh in 2 seconds\nskip cleanup, not a docker build: 0b59a18ab905\ninstalling v2 tool kustomize v5.0.0\nlinking tool kustomize v5.0.0\nv5.0.0\nInstalled v2 /usr/local/buildpack/tools/v2/kustomize.sh in 3 seconds\nskip cleanup, not a docker build: 0b59a18ab905\n",
       "stderr": "Adding repo incubator https://charts.helm.sh/incubator/\n\"incubator\" has been added to your repositories\n\nUpdating dependency /tmp/chartify1809371219/raw1/raw\nUpdating dependency /tmp/2864070368\nHang tight while we grab the latest from your chart repositories...\n...Successfully got an update from the \"incubator\" chart repository\nUpdate Complete. ⎈Happy Helming!⎈\nSaving 1 charts\nDownloading raw from repo https://charts.helm.sh/incubator/\nDeleting outdated charts\n\n"
DEBUG: Updated 1 package files (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Updated 1 lock files (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "updatedArtifacts": ["helmfile.lock"]
DEBUG: Getting comments for #1 (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Found 0 comments (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: isBranchConflicted(main, renovate/raw-0.x) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: branch.isConflicted(): using git to calculate (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Setting git author name: Renovate Bot (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Setting git author email: renovate@whitesourcesoftware.com (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: branch.isConflicted(): false (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: 2 file(s) to commit (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Preparing files for committing to branch renovate/raw-0.x (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: git commit (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "deletedFiles": [],
       "ignoredFiles": [],
       "result": {
         "author": null,
         "branch": "renovate/raw-0.x",
         "commit": "03b8b4b4fc3e7736d2a82da591b6aa26db1f0316",
         "root": false,
         "summary": {"changes": 2, "insertions": 5, "deletions": 5}
       }
DEBUG: Pushing refSpec renovate/raw-0.x:renovate/raw-0.x (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: git push (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "result": {
         "pushed": [],
         "ref": {"local": "refs/remotes/origin/renovate/raw-0.x"},
         "remoteMessages": {"all": []}
       }
DEBUG: Setting current branch to main (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: latest commit (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "branchName": "main",
       "latestCommitDate": "2023-03-13T12:24:53+01:00"
 INFO: Branch updated (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "commitSha": "03b8b4b4fc3e7736d2a82da591b6aa26db1f0316"
DEBUG: Ensuring PR (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: There are 0 errors and 0 warnings (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: getBranchPr(renovate/raw-0.x) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: findPr(renovate/raw-0.x, undefined, open) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Found PR #1 (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: getPrCache() (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Found existing PR (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Fetching changelog: https://github.com/helm/charts (0.1.0 -> 0.2.5) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: No Github tags found for repository:helm/charts (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: Processing existing PR (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: updatePr(1, Update Helm release raw to v0.2.5, body) (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: PR updated...prNo: 1 (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
 INFO: PR updated (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
       "pr": 1,
       "prTitle": "Update Helm release raw to v0.2.5"
DEBUG: setPrCache() (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: PR is not configured for automerge (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: setBranchCommit() (repository=nikolaik/test-renovate-helmfile-kustomize, branch=renovate/raw-0.x)
DEBUG: getBranchPr(renovate/raw-0.x) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: findPr(renovate/raw-0.x, undefined, open) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Found PR #1 (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: branch.isBehindBase(): using cached result "false" (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: isBranchConflicted(main, renovate/raw-0.x) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: branch.isConflicted(): using cached result "false" (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: getPrCache() (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Ensuring Dependency Dashboard (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: ensureIssue(Dependency Dashboard) (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Issue is open and up to date - nothing to do (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Removing any stale branches (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: config.repoIsOnboarded=true (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Branch lists (repository=nikolaik/test-renovate-helmfile-kustomize)
       "branchList": ["renovate/raw-0.x"],
       "renovateBranches": ["renovate/raw-0.x"]
DEBUG: remainingBranches= (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: No branches to clean up (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: PackageFiles.clear() - Package files deleted (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Branch summary (repository=nikolaik/test-renovate-helmfile-kustomize)
       "cacheModified": undefined,
       "baseBranches": [{"branchName": "main", "sha": "acfea149bebe380e16c936cc29a8ede9d398ba23"}],
       "branches": [
         {
           "automerge": false,
           "baseBranch": "main",
           "baseBranchSha": "acfea149bebe380e16c936cc29a8ede9d398ba23",
           "branchName": "renovate/raw-0.x",
           "branchSha": "03b8b4b4fc3e7736d2a82da591b6aa26db1f0316",
           "isModified": false,
           "isPristine": true
         }
       ],
       "inactiveBranches": []
DEBUG: Renovate repository PR statistics (repository=nikolaik/test-renovate-helmfile-kustomize)
       "stats": {"total": 1, "open": 1, "closed": 0, "merged": 0}
DEBUG: Repository result: done, status: onboarded, enabled: true, onboarded: true (repository=nikolaik/test-renovate-helmfile-kustomize)
DEBUG: Repository timing splits (milliseconds) (repository=nikolaik/test-renovate-helmfile-kustomize)
       "splits": {"init": 3590, "extract": 723, "lookup": 267, "onboarding": 0, "update": 22878},
       "total": 27780
DEBUG: Package cache statistics (repository=nikolaik/test-renovate-helmfile-kustomize)
       "get": {"count": 18, "avgMs": 0, "medianMs": 0, "maxMs": 1},
       "set": {"count": 18, "avgMs": 75, "medianMs": 5, "maxMs": 936}
DEBUG: http statistics (repository=nikolaik/test-renovate-helmfile-kustomize)
       "urls": {
         "https://api.github.com/graphql (POST,200)": 10,
         "https://api.github.com/repos/helm/charts (GET,200)": 1,
         "https://api.github.com/repos/helm/charts/git/trees/master (GET,200)": 1,
         "https://api.github.com/repos/nikolaik/test-renovate-helmfile-kustomize/issues/1/comments (GET,200)": 1,
         "https://api.github.com/repos/nikolaik/test-renovate-helmfile-kustomize/issues/2 (GET,200)": 2,
         "https://api.github.com/repos/nikolaik/test-renovate-helmfile-kustomize/pulls (GET,200)": 1,
         "https://api.github.com/repos/nikolaik/test-renovate-helmfile-kustomize/pulls/1 (PATCH,200)": 1,
         "https://charts.helm.sh/incubator/index.yaml (GET,200)": 1
       },
       "hostStats": {
         "api.github.com": {"requestCount": 17, "requestAvgMs": 448, "queueAvgMs": 0},
         "charts.helm.sh": {"requestCount": 1, "requestAvgMs": 59, "queueAvgMs": 0}
       },
       "totalRequests": 18
DEBUG: Package lookup durations (repository=nikolaik/test-renovate-helmfile-kustomize)
       "helm": {"count": 1, "averageMs": 218, "totalMs": 218, "maximumMs": 218}
DEBUG: dns cache (repository=nikolaik/test-renovate-helmfile-kustomize)
       "hosts": []
 INFO: Repository finished (repository=nikolaik/test-renovate-helmfile-kustomize)
       "cloned": true,
       "durationMs": 27780
DEBUG: Checking file package cache for expired items
DEBUG: Deleted 0 of 18 file cached entries in 9ms
DEBUG: Renovate exiting
Done in 35.24s.

```

viceice commented on lib/modules/manager/helmfile/artifacts.ts:
```diff
@@ -54,6 +54,15 @@ export async function updateArtifacts({
+      (dep) => dep.managerData?.needKustomize
+    );
+    if (needKustomize) {
+      execOptions.toolConstraints?.push({
```
```suggestion
      execOptions.toolConstraints!.push({
```
you should store the constraints array as separate variable, so you don't need the assertion.
